src/login/pam_systemd.c

```diff
@@ -948,7 +948,7 @@ _public_ PAM_EXTERN int pam_sm_open_session(
 
         /* Talk to logind over the message bus */
 
-        r = pam_acquire_bus_connection(handle, &bus);
+        r = pam_acquire_bus_connection(handle, "system-bus-pam-systemd", &bus);
```

Comment:
i'd really just pass the module name here, and let pam_acquire_bus_connection() prefix it with "system-bus-"

i am pretty sure we should include the pid too in this, so that we never are tempted to share a bus object after fork(), and hence the name should be joined together in pam_acquire_bus_connection() fully

Reply:
> i'd really just pass the module name here, and let pam_acquire_bus_connection() prefix it with "system-bus-"

Done

> i am pretty sure we should include the pid too in this, so that we never are tempted to share a bus object after fork(), and hence the name should be joined together in pam_acquire_bus_connection() fully

Given the very scarce testing coverage, I prefer to leave that for later, and do it via sd-bus directly, so that it's not duplicated. And most importantly, so that I can backport these commits to systemd-stable with lower risk, to fix exactly the issue at hand and avoid speculative changes. We can do this via sd-bus later in main, and let it simmer for a while, and avoiding to backport it to stable.